draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -548,6 +592,92 @@ Elligator2 can be simplified with projective coordinates.
 
 ((TODO: write this variant))
 
+## FT-hash
+
+{{FT12}} outline a method for hashing onto a set of curves which includes BN curves, which are suitable
+for bilinear pairings. A number of cryptographic schemes rely on pairings, including some identity-based
+encryption (IBE) and attribute-based encryption (ABE) schemes.  BN curves are not one specific curve but a
+class of curves which can be constructed using Barreto and Naehrig's algorithm {{BN05}}. BN curves are a
+preferred option for asymmetric pairings as they provide relatively efficient computation for pairings, which are
+usually the slowest computation in any cryptosystem using them. There is also work giving parameter selection
+for 256-bit BN-curves, resulting in 128 bits of security for hashing onto BN curves. Suggestions for suitable BN
+curves can be found in {{draft-BN}} (work in progress).
+
+This method is suitable for BN-curves, which have the form E: f(x) = y^2 = x^3 + b, where operations
+are over a field Fp,  where p = 7 (mod 12) and 1 + b is a nonzero square in Fp. Fouque and Tibouchi give an 
+encoding F which makes the hash function Hash(m) = F(H1(m)) + F(H2(m)) indifferentiable from a random oracle,
+where  H1 and H2 are independent hash functions into Fp. We give this encoding and hash function construction
+in the steps below.  In the following, Xp() is the Legendre symbol (Xp(0) = 0, for a != 0 Xp(a) = 1 if a is a square in
```

Comment:
If we're using Xp() for the Legendre symbol, which is fine by me, we should re-write the other algorithms to make use of this notation.